cxx/gendb.hh

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+#include <map>
+#include <random>
+#include <string>
+
+#include "distributions/crp.hh"
+#include "hirm.hh"
+#include "observations.hh"
+#include "pclean/schema.hh"
+#include "pclean/schema_helper.hh"
+
+class GenDB {
+ public:
+  const PCleanSchema& schema;
+
+  // TODO(emilyaf): Merge PCleanSchemaHelper and GenDB.
+  PCleanSchemaHelper schema_helper;
+
+  // This data structure contains entity sets and linkages. Semantics are
+  // map<class_name, map<primary_key, map<reference_field_name, ref_val>>>,
+  // where primary_key and ref_val are (integer) entity IDs.
+  std::map<std::string, std::map<int, std::map<std::string, int>>>
+      reference_values;
+
+  HIRM* hirm;
+
+  // CRPs for latent entities, where the "tables" are entity IDs and the
+  // "customers" are unique identifiers of observations of that class. Map
+  // keys are class names.
+  std::map<std::string, CRP> domain_crps;
+
+  GenDB(std::mt19937* prng, const PCleanSchema& schema,
+        bool _only_final_emissions = false, bool _record_class_is_clean = true);
+
+  void incorporate(
```

Comment:
All of these methods need descriptive comments.

Reply:
Done.